packages/repeater/src/request-runner/protocols/HttpRequestRunner.ts

```diff
@@ -0,0 +1,244 @@
+      options.method,
+      options.url
+    );
+    this.logger?.error('Cause: %s', message);
+
+    return new Response({
+      protocol: this.protocol,
+      message,
+      errorCode
+    });
+  }
+
+  private async request(options: Request): Promise<IncomingResponse> {
+    let responseTruncatePromise: Promise<Buffer | undefined>;
+
+    const res = await request({
+      agent: this.getRequestAgent(options),
+      body: options.body,
+      followRedirect: false,
+      gzip: true,
+      method: options.method,
+      resolveWithFullResponse: true,
+      strictSSL: false,
+      rejectUnauthorized: false,
+      timeout: this.options.timeout,
+      url: options.url
+    })
+      .on('request', (req: OutgoingMessage) => this.setHeaders(req, options))
+      .on(
+        'response',
+        (response: IncomingResponse) =>
+          (responseTruncatePromise = this.truncateResponse(response))
+      );
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const body = await responseTruncatePromise!;
+
+    if (body) {
+      res.body = body.toString();
+      res.headers['content-length'] = String(body.byteLength);
+    }
+
+    return res;
+  }
+
+  private getRequestAgent(options: Request) {
+    return (
+      this.proxy ??
+      (options.url.startsWith('https') ? this.httpsAgent : this.httpAgent)
+    );
+  }
+
+  private async truncateResponse(
+    res: IncomingResponse
+  ): Promise<Buffer | undefined> {
+    if (res.statusCode === 204 || res.method === 'HEAD') {
+      return;
+    }
+
+    const type = this.parseContentType(res);
+    const maxBodySize = this.maxContentLength * 1024;
+
+    const requiresTruncating = !this.options.whitelistMimes?.some(
+      (mime: string) => type.startsWith(mime)
+    );
+
+    const body = await this.parseBody(res, { maxBodySize, requiresTruncating });
+
+    return body;
+  }
+
+  private parseContentType(res: IncomingResponse): string {
+    let type = res.headers['content-type'] || 'text/plain';
+
+    try {
+      ({ type } = contentTypeParse(type));
+    } catch {
+      // noop
+    }
+
+    return type;
+  }
+
+  private async parseBody(
+    res: IncomingResponse,
+    options: { maxBodySize: number; requiresTruncating: boolean }
+  ): Promise<Buffer> {
+    let truncated = false;
+
+    const chunks = [];
+    let chunksByteLength = 0;
+
+    for await (const chunk of res) {
+      chunks.push(chunk);
+
+      chunksByteLength += Buffer.from(chunk).byteLength;
+      truncated =
+        this.maxContentLength !== -1 &&
+        chunksByteLength > options.maxBodySize &&
+        options.requiresTruncating;
+
+      if (truncated) {
+        res.destroy();
+        break;
+      }
+    }
+
+    if (truncated) {
+      this.logger?.debug(
+        'Truncate original response body to %i bytes',
+        options.maxBodySize
+      );
+
+      return Buffer.concat(chunks).slice(0, options.maxBodySize);
+    }
+
+    return Buffer.concat(chunks);
```

Comment:
I guess we can extract a local variable
```suggestion
    const body = Buffer.concat(chunks);
    
    if (truncated) {
      this.logger?.debug(
        'Truncate original response body to %i bytes',
        options.maxBodySize
      );

      return body.slice(0, options.maxBodySize);
    }

    return body;
```